common.bazelrc

```diff
@@ -42,16 +42,17 @@ test --test_output=errors
 # --define=VERBOSE_LOGS=1
 #     Rules will output verbose logs if the VERBOSE_LOGS environment variable is set. `VERBOSE_LOGS` will be passed to
 #     `nodejs_binary` and `nodejs_test` via the default value of the `default_env_vars` attribute of those rules.
-# --define=DEBUG=1
-#     Rules may change their build outputs if the DEBUG environment variable is set. For example,
-#     mininfiers such as terser may make their output more human readable when this is set. `DEBUG` will be passed to
+# --compilation_mode=dbg
+#     Rules may change their build outputs if the compilation mode is set to dbg. For example,
+#     mininfiers such as terser may make their output more human readable when this is set. `COMPILATION_MODE` will be passed to
 #     `nodejs_binary` and `nodejs_test` via the default value of the `default_env_vars` attribute of those rules.
+#     See https://docs.bazel.build/versions/master/user-manual.html#flag--compilation_mode for more details.
 test:debug --test_output=streamed --test_strategy=exclusive --test_timeout=9999 --nocache_test_results --define=VERBOSE_LOGS=1
 # Use bazel run with `--config=debug` to turn on the NodeJS inspector agent.
 # The node process will break before user code starts and wait for the debugger to connect.
 run:debug --define=VERBOSE_LOGS=1 -- --node_options=--inspect-brk
 # The following option will change the build output of certain rules such as terser and may not be desirable in all cases
-build:debug --define=DEBUG=1
+build:debug --compilation_mode=dbg
```

Comment:
I think we should just remove this:
- it's confusing that `--config=debug` does something for test and a different thing for build
- `-c=debug` is short so the syntax sugar has no benefit

Reply:
SGTM. I can send another CL later, as I want this change to be a drop in replacement for `DEBUG` usages, without messing too much about tests and CI configs.

Also, I haven’t tried if `-c debug` works, as the mode seems always being referred to as `dbg` and `opt`, which also matches the value of `COMPILATION`_MODE context variable.

Reply:
We should not use `--compilation_mode=dbg` flag (also it's alias `-c dbg`), this flag only affects C/C++ build.
See https://docs.bazel.build/versions/master/user-manual.html#flag--compilation_mode

Reply:
Sorry for commenting without really looking into the PR. I believe this PR is fine.

`--compilation_mode` was designed for C++, but that doesn't mean we cannot use it for other languages. The value of `--compilation_mode` is exported as `COMPILATION_MODE` variable. Since this PR is actually using the information to configure the build, I think it's totally fine.

Reply:
BTW, do we have test to make sure this actually work?

Reply:
Yes. It’s all covered. Tests are not changed to assure that the behavior is consistent.